applications/OptimizationApplication/custom_utilities/filtering/explicit_filter.cpp

```diff
@@ -126,37 +126,60 @@ void ComputeWeightForAllNeighbors(
     double& rSumOfWeights,
     std::vector<double>& rListOfWeights,
     const FilterFunction& rFilterFunction,
-    const double VertexMorphingRadius,
+    const double Radius,
     const EntityPoint<TEntityType>& rDesignPoint,
     const std::vector<typename EntityPoint<TEntityType>::Pointer>& rNeighbourNodes,
+    const std::vector<typename EntityPoint<TEntityType>::Pointer>& rDampingNeighbourNodes,
     const IndexType NumberOfNeighbours,
     Expression const * const pExpression)
 {
     for (IndexType i = 0; i < NumberOfNeighbours; ++i) {
         const double domain_size = GetDomainSize(*rNeighbourNodes[i], pExpression);
-        const double filter_weight = rFilterFunction.ComputeWeight(rDesignPoint.Coordinates(), rNeighbourNodes[i]->Coordinates(), VertexMorphingRadius);
-
-        rListOfWeights[i] = TWeightComputationType::Compute(filter_weight, domain_size);
+        const double filter_weight = rFilterFunction.ComputeWeight(rDesignPoint.Coordinates(), rNeighbourNodes[i]->Coordinates(), Radius);
+        double damping_multiplier = 1.0;
+        if(rDampingNeighbourNodes.size()>0){
+            auto dist_vector = rNeighbourNodes[i]->Coordinates() - rDampingNeighbourNodes[i]->Coordinates();
```

Comment:
```suggestion
            const array_1d<double, 3>& dist_vector = rNeighbourNodes[i]->Coordinates() - rDampingNeighbourNodes[i]->Coordinates();
```
It is dangerous to use auto with `array_1d<double,3>`, `Vector` or  `Matrix` when they are doing some computations (such as substraction in this case). Because, they are compiled differently in different compilers (such as clang and intel) and produces different results. The reasoning is, they are compile time expressions which are deduced differently. It is safer to tell the final type instead of using `auto`